### ISSUE TITLE
Export of bursts not following template rule

### ISSUE THREAD

**Issue by neebah** [RhetTbull/osxphotos#401]
I've organized all of my photos into albums (including the bursts).  When I run an export using the {folder_album} template one of the bursts is put in the album it was assigned and the others are place in the (-) folder for some reason.

**Comment by RhetTbull** [comment]
I'll need to think about how to handle this case.  It's reasonable to assume the burst photos would be exported into the same album folder as the key photo.  However, technically, the burst photos are *not* in the album, only the key photo is.  Until you select burst images with the "Make a selection" button, they're in sort of a limbo state in Photos' database.  For example, if you export a burst image from Photos, it exports only the selected key image, not all the associated burst photos.  On Catalina at least, if you add a burst photo to an album, you will see it marked as burst, but when viewing the album, you'll no longer see the "Make a selection" button on the photo.  

So if osxphotos exported all bursts to the same album as the key feature, this would be different than what Photos does and I've tried to mirror what Photos does in most cases.  On the other hand, this could be seen as unexpected behavior -- you clearly expected the burst images to export to the same folder as the key image and this is entirely reasonable.  I might be able to add a flag like "--export-bursts-to-album" or something like that to get the behavior you're expecting.  This will take some work because I'll have to add logic to associate the bursts with the right album -- the Photos database does not associate the bursts with the album.

**Comment by RhetTbull** [comment]
Also, the "_" folder is the default for anything that doesn't match a template. You can change the default value by specifying it after a comma following the template field as in: "{folder_album,NoAlbum}"

**Comment by neebah** [comment]
When I've run Export Unmodified originals in the past all parts of the burst have exported.

**Comment by neebah** [comment]
So I just did a test by taking a burst on my phone and putting it into an album on the Photos app.  I then exported unmodified originals (using the photos app) and chose album name with number.  These exported with the correct name of the album.  So there must be some data in the photos database to link the bursts with the albums that they are put in.

**Comment by RhetTbull** [comment]
Thanks. The burst photos are not associated with the album but they are associated with the photo in the album.  I'll have to add a check when exporting something along the lines of:

If photo is burst:
    Get list of all other photos in burst set
     For each of these, get list of albums they are in
     Use these album names to fill in template

**Comment by RhetTbull** [comment]
Need to update this:

https://github.com/RhetTbull/osxphotos/blob/de94fd76deef84c6446f705071c40f05b787f653/osxphotos/phototemplate.py#L854-L875

and this:
https://github.com/RhetTbull/osxphotos/blob/de94fd76deef84c6446f705071c40f05b787f653/osxphotos/phototemplate.py#L842-L843

Add `PhotoInfo.burst_albums`, `PhotoInfo.burst_album_info`

**Comment by RhetTbull** [comment]
I have a fix implemented but in the process discovered two new bugs related to burst photos (#403 and #404) so I am working to fix those in the same release.

**Comment by RhetTbull** [comment]
I think this is fixed in [v0.41.5](https://github.com/RhetTbull/osxphotos/releases/tag/v0.41.5).  Let me know if you still have trouble with this issue.

**Comment by RhetTbull** [comment]
Re-opening this to think some more about what osxphotos should do:

Here's what Photos actually does:

- When you take a burst photo, the photo you see on top of the "burst stack" is the "key photo".  If you add this photo to an album, all associated burst photos in the burst set are also added to the album and if you "Export Unmodified Original" then Photos exports all the images in the burst set.
- If you make a selection of burst photos using the "Make a Selection" button, photos adds the selected images as separate images to the Library.  If you add one of these to an album, Photos does *not* add the rest of the photos from the burst set to the album and if you export one of these selected photos, the other images from the burst set are *not* exported with "Export Unmodified Original".

**Comment by RhetTbull** [comment]
Making some progress on decoding burst types with more fidelity.  

the "pick type" seems to be a bit flag indicating status of images in the burst set:

- In Photos <=4, this value is in `RKVersion.burstPickType`
- In Photos >= 5, this value is in `ZGENERICASSET.ZAVALANCHEPICKTYPE` (`ZASSET` table in Photos 6)

The values appear to be:

```
BURST_NOT_SELECTED = 0b10 # 2: burst image is not selected
BURST_SELECTED = 0b1000 # 8: burst image is selected
BURST_KEY = 0b10000 # 16: burst image is the key photo (top of burst stack)
BURST_DEFAULT_PICK = 0b100 # 4: burst image is the one Photos picked to be key image before any selections made
BURST_UNKNOWN = 0b100000 # 32: this is almost always set with BURST_DEFAULT_PICK and never if BURST_DEFAULT_PICK is not set.  I think this has something to do with what algorithm Photos used to pick the default image
```

Thus, an unselected burst image has a value of 2.  
A selected image could be 

- 8: BURST_SELECTED
- 12: BURST_SELECTED & BURST_DEFAULT_PICK
- 44: BURST_SELECTED & BURST_DEFAULT_PICK & BURST_UNKNOWN

A key image could be:
- 16: BURST_KEY
- 20: BURST_KEY & BURST_DEFAULT_PICK
- 52: BURST_KEY & BURST_DEFAULT_PICK & BURST_UNKNOWN

**Comment by RhetTbull** [comment]
I think this is actually fixed correctly in 0.42.11.  osxphotos now does what Photos does when using "Export Unmodified Originals".  If the burst "key image" is in an album, all photos associated with the burst set get exported to that album.  If a "selected burst image" that's not the "key image" is in an album, only that photo gets exported -- it's not treated as part of the burst set.

**Comment by RhetTbull** [comment]
This seems to have regressed.  Burst images are not getting correctly exported.
